### ISSUE TITLE
Error when combining animatedStyle with className on mobile in React Native + Expo app

### ISSUE THREAD

**Issue by JeanDes-Code** [nativewind/nativewind#846]
**Describe the bug**
When building a React Native app with Expo, using NativeWind and React Native Reanimated, applying both an animated style from React Native Reanimated and a className from NativeWind to the same element causes an error on mobile versions. The error occurs only after a hot reload and is as follows: 
`"Error: You attempted to set the key 'translateX' with the value '0' on an object that is meant to be immutable and has been frozen." `
This issue does not affect the web version, only mobile.

**To Reproduce**

1. Create a React Native + Expo app using the specified versions ("react-native": "0.73.2", "expo": "~50.0.4", "react-native-reanimated": "^3.6.2").
2. Apply an animatedStyle from React Native Reanimated and a className from NativeWind to the same element.
3. Observe the app functioning correctly on the first render.
4. Trigger a hot reload by updating and saving code while dev app is opened on emulator or on your mobile device via expo go .
5. Notice the error regarding an attempt to set a key on a frozen object.

**Expected behavior**
The expectation was that combining an animated style with a NativeWind className would not cause any issues, allowing for both animations and Tailwind CSS styles to be applied to the same element seamlessly on mobile devices.

**Additional context**
The problem seems to stem from the way React Native Reanimated creates a frozen object for the animated style, which then conflicts with NativeWind's styling application process for native mobile devices. The workaround discovered is to avoid applying both types of styles to the same element when targeting native mobile devices. This limitation might not be widely known, and documenting it could help others encountering the same issue.

Though it might not be feasible to eliminate this limitation due to the inherent nature of the libraries involved, raising awareness about this issue could be beneficial for developers using these technologies together. 
I appreciate the work being done on NativeWind and hope this report helps improve the experience for others. ✌️

**Comment by kimchouard** [comment]
Interesting... Seems like animating via a reanimated style and have NativeWind classes at the same time is a pretty common use case.

Unless there is a different approach to animating a View with Reanimated while using NativeWind @marklawlor ?

@JeanDes-Code Can you share a code snippet as an example? :)

**Comment by JeanDes-Code** [comment]
Hey @kimchouard, here is a code snippet :

```tsx
export const Button = ({
  ...
}: ButtonProps) => {

  let loaderXPos = useSharedValue(-30);
  let loaderOpacity = useSharedValue(0);
  let loaderRotation = useSharedValue(0);
  
  const loaderAnimatedStyle = useAnimatedStyle(() => {
    return {
      transform: [
        { translateX: loaderXPos.value },
        { rotate: `${loaderRotation.value}deg` },
      ] as [{ translateX: number }, { rotate: string }],
      opacity: loaderOpacity.value as number,
    };
  });
  
  return (
  
    <Animated.View
        style={loaderAnimatedStyle}
        className="flex items-center justify-center"
        >
          <LoaderSvg
             color="white"
             size={loaderSize}
           />
     </Animated.View>
   )
}
```

My workaround was to do something like that : 


```tsx
<Animated.View style={loaderAnimatedStyle}>
    <View className='flex items-center justify-center'>
         <LoaderSvg
            color='white'
            size={loaderSize}
            />
      </View>
</Animated.View>
```

**Comment by zigcccc** [comment]
+1 also experiencing this. The suggested workaround does the job, but it would be a lot cleaner to be able to merge the two together.

**Comment by danstepanov** [comment]
We are closing all pre-v4.1 issues now that v4.1 is released. If the problem is still happening with the latest v4.1, [please open a new issue](https://github.com/nativewind/nativewind/blob/next/contributing.md#opening-an-issue) and reference this one in the description.
